### PR TITLE
[DCA-37] Add CNAME for dca-test application

### PR DESCRIPTION
**Pull request by jesusaurus**
Now that we have created a wildcard certificate for our application subdomain and are using it in our application, set up a CNAME redirect so that the application domain matches the certificate domain.

Depends on: https://github.com/Sage-Bionetworks/data_curator-infra/pull/29

### CODE REVIEW

File: org-formation/800-redirects/_tasks.yaml [hunk 1]
```diff
@@ -35,3 +37,22 @@ FinopsApiRedirect:  # redirect finops-api.sageit.org to lambda-mips-api cloudfro
     SourceHostedZoneId: "Z0478495257GEB73WFM63"
     # the value of the CNAME record
     TargetHostName: !CopyValue [!Sub '${primaryRegion}-${resourcePrefix}-mips-microservice-CloudfrontDomain', !Ref AdminCentralAccount]
+
+
+# *.app.sagebionetworks.org
+
+DcaDevAppRedirect:  # redirect dca-test.app.sagebionetworks.org to data_curator-infra ALB
```
Comment (brucehoff) [inline org-formation/800-redirects/_tasks.yaml:44]: @jesusaurus can you confirm that, though you use the term 'redirect' there isn't literally an HTTP redirect (3xx response), but rather traffic is forwarded to the target ALB?

Comment (jesusaurus) [reply org-formation/800-redirects/_tasks.yaml:44]: @brucehoff correct, this is not an HTTP redirect, the redirection is happening when looking up the IP address of the host name (which I have an old habit of also calling a redirect, but I can change that terminology if its confusing)

File: org-formation/800-redirects/_tasks.yaml [hunk 2]
```diff
@@ -35,3 +37,22 @@ FinopsApiRedirect:  # redirect finops-api.sageit.org to lambda-mips-api cloudfro
...
+  Type: update-stacks
+  Template: https://raw.githubusercontent.com/Sage-Bionetworks/aws-infra/v0.6.4/templates/R53/cname.yaml
+  StackName: !Sub '${resourcePrefix}-finops-api-redirect'
+  StackDescription: Setup a redirect to the lambda-mips-api cloudfront distribution
+  DefaultOrganizationBindingRegion: !Ref primaryRegion
+  DefaultOrganizationBinding:
+    Account: !Ref SageITAccount
+  Parameters:
+    # the name of the CNAME record
+    SourceHostName: "dca-test.app.sagebionetworks.org"
```
Comment (zaro0508) [inline org-formation/800-redirects/_tasks.yaml:54]: this is more likely to be dca-dev / dca-staging / dca-prod

File: org-formation/800-redirects/_tasks.yaml [hunk 3]
```diff
@@ -35,3 +37,22 @@ FinopsApiRedirect:  # redirect finops-api.sageit.org to lambda-mips-api cloudfro
...
     # the value of the CNAME record
     TargetHostName: !CopyValue [!Sub '${primaryRegion}-${resourcePrefix}-mips-microservice-CloudfrontDomain', !Ref AdminCentralAccount]
+
+
+# *.app.sagebionetworks.org
+
+DcaDevAppRedirect:  # redirect dca-test.app.sagebionetworks.org to data_curator-infra ALB
+  Type: update-stacks
+  Template: https://raw.githubusercontent.com/Sage-Bionetworks/aws-infra/v0.6.4/templates/R53/cname.yaml
+  StackName: !Sub '${resourcePrefix}-finops-api-redirect'
```
Comment (zaro0508) [inline org-formation/800-redirects/_tasks.yaml:47]: finops?

File: org-formation/800-redirects/_tasks.yaml [hunk 4]
```diff
@@ -35,3 +37,22 @@ FinopsApiRedirect:  # redirect finops-api.sageit.org to lambda-mips-api cloudfro
...
+  StackDescription: Setup a redirect to the lambda-mips-api cloudfront distribution
```
Comment (zaro0508) [inline org-formation/800-redirects/_tasks.yaml:48]: mips?

File: org-formation/800-redirects/_tasks.yaml (hunk 1 above)
Comment (zaro0508) [reply org-formation/800-redirects/_tasks.yaml:44]: could we add a comment referencing the GH repo for this redirect?

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: org-formation/800-redirects/_tasks.yaml (hunk 1 above)
Comment (jesusaurus) [reply org-formation/800-redirects/_tasks.yaml:44]: @zaro0508 done

File: org-formation/800-redirects/_tasks.yaml (hunk 2 above)
Comment (jesusaurus) [reply org-formation/800-redirects/_tasks.yaml:54]: good call, I've updated this name to align with the context name

File: org-formation/800-redirects/_tasks.yaml (hunk 3 above)
Comment (jesusaurus) [reply org-formation/800-redirects/_tasks.yaml:47]: whoops, I guess I needed a little more coffee this morning

File: org-formation/800-redirects/_tasks.yaml (hunk 4 above)
Comment (jesusaurus) [reply org-formation/800-redirects/_tasks.yaml:48]: ditto

**Review by zaro0508 — APPROVED** [review]

**Pull request merged**